uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/Service/LeaveRequestRights.php

```diff
@@ -209,4 +209,35 @@ public function getLeaveContactsCurrentUserHasAccessTo() {
 
     return $results;
   }
+
+  /**
+   * Checks if the current user can cancel the leave request for the given absence type,
+   * leave contact and leave from date.
+   *
+   * @param int $absenceTypeId
+   * @param int $contactId
+   * @param \DateTime $leaveFromDate
+   *
+   * @return bool
+   */
+  public function canCancelForAbsenceType($absenceTypeId, $contactId, DateTime $leaveFromDate) {
+    if ($this->currentUserIsAdmin() || $this->currentUserIsLeaveManagerOf($contactId)) {
+      return TRUE;
+    }
+
+    $absenceType = AbsenceType::findById($absenceTypeId);
+    if ($absenceType->allow_request_cancelation == AbsenceType::REQUEST_CANCELATION_ALWAYS) {
+      return TRUE;
+    }
+
+    $today = new DateTime('today');
+
+    if ($absenceType->allow_request_cancelation != AbsenceType::REQUEST_CANCELATION_NO &&
```

Comment:
This is a bit long, and also a bit confusing to read, do you think you can make 

`$absenceType->allow_request_cancelation != AbsenceType::REQUEST_CANCELATION_NO` into a descriptive variable?